src/tests/ftest/osa/osa_online_parallel_test.py

```diff
@@ -159,33 +148,27 @@ def ior_thread(self, pool, oclass, api, test, flags, results):
         except CommandFailure as _error:
             results.put("FAIL")
 
-    def dmg_thread(self, puuid, rank, target, action, results):
+    def dmg_thread(self, action, action_args, results):
         """Generate different dmg command related to OSA.
             Args:
-            puuid (int) : Pool UUID
-            rank (int)  : Daos Server Rank
-            target (list) : Target list
-            action (string) : String to identify OSA action
-                              like drain, reintegration, extend
+            action_args(dict) : {action: {"puuid":
+                                          pool[val].uuid,
+                                          "rank": rank,
+                                          "target": t_string,
+                                          "action": action,}
             results (queue) : dmg command output queue.
         """
         # Give sometime for IOR threads to start
         dmg = copy.copy(self.dmg_command)
         try:
-            if action == "exclude":
-                dmg.pool_exclude(puuid, (rank + 1), target)
-            elif action == "drain":
-                dmg.pool_drain(puuid, rank)
-            elif action == "reintegrate":
+            if action == "reintegrate":
                 time.sleep(60)
-                dmg.pool_reintegrate(puuid, (rank + 1), target)
-            else:
-                self.fail("Invalid action for dmg thread")
+            getattr(dmg, "pool_{}".format(action))(**action_args[action])
         except CommandFailure as _error:
-            results.put("{} failed".format(action))
-            # Future enhancement for extend
-            # elif action == "extend":
-            #    dmg.pool_extend(puuid, (rank + 2))
+            results.put("{} failed".format(action_args[action]))
+        # Future enhancement for extend
+        # elif action == "extend":
+        #    dmg.pool_extend(puuid, (rank + 2))
```

Comment:
This comment line should be updated. I think you need to let the caller know how to create the action_args for this case.

Reply:
Will fix these issue while I re-enable the test. Presently, the test is skipped due to existing bug DAOS-5877. Thanks for your input

Reply:
done.